specification/guestconfiguration/GuestConfiguration.Management/models.tsp

```diff
@@ -0,0 +1,801 @@
+  uuid?: string;
+}
+
+/**
+ * The guest configuration assignment resource.
+ */
+model AssignmentReportResource {
+  /**
+   * A value indicating compliance status of the machine for the assigned guest configuration.
+   */
+  @visibility(Lifecycle.Read)
+  complianceStatus?: ComplianceStatus;
+
+  /**
+   * Name of the guest configuration assignment resource setting.
+   */
+  @visibility(Lifecycle.Read)
+  resourceId?: string;
+
+  /**
+   * Compliance reason and reason code for a resource.
+   */
+  @OpenAPI.extension("x-ms-identifiers", #["phrase", "code"])
+  reasons?: AssignmentReportResourceComplianceReason[];
+
+  /**
+   * Properties of a guest configuration assignment resource.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-empty-model" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  properties?: {};
+}
+
+/**
+ * Reason and code for the compliance of the guest configuration assignment resource.
+ */
+model AssignmentReportResourceComplianceReason {
+  /**
+   * Reason for the compliance of the guest configuration assignment resource.
+   */
+  @visibility(Lifecycle.Read)
+  phrase?: string;
+
+  /**
+   * Code for the compliance of the guest configuration assignment resource.
+   */
+  @visibility(Lifecycle.Read)
+  code?: string;
+}
+
+/**
+ * Information about VMSS VM
+ */
+model VmssvmInfo {
+  /**
+   * UUID of the VM.
+   */
+  @visibility(Lifecycle.Read)
+  vmId?: string;
+
+  /**
+   * Azure resource Id of the VM.
+   */
+  @visibility(Lifecycle.Read)
+  vmResourceId?: string;
+
+  /**
+   * A value indicating compliance status of the machine for the assigned guest configuration.
+   */
+  @visibility(Lifecycle.Read)
+  complianceStatus?: ComplianceStatus;
+
+  /**
+   * Id of the latest report for the guest configuration assignment.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  latestReportId?: string | null;
+
+  /**
+   * Date and time when last compliance status was checked.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read)
+  lastComplianceChecked?: utcDateTime | null;
+}
+
+/**
+ * The core properties of ARM resources
+ */
+model Resource {
+  /**
+   * ARM resource id of the guest configuration assignment.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  /**
+   * Name of the guest configuration assignment.
+   */
+  name?: string;
+
+  /**
+   * Region where the VM is located.
+   */
+  location?: string;
+
+  /**
+   * The type of the resource.
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model ErrorResponseError {
```

Comment:
>ErrorResponseError [](http://example.com/codeflow?start=6&length=18)

We shouldn't be redeclaring ErrorResponse from common types, right.

Whats the justification?